interfaces/mount/spec.go

```diff
@@ -39,10 +46,103 @@ func (spec *Specification) AddMountEntry(e Entry) error {
 	return nil
 }
 
+// resolveSpecialVariable resolves $SNAP, $SNAP_DATA and $SNAP_COMMON.
+func resolveSpecialVariable(path string, snapInfo *snap.Info) string {
+	return os.Expand(path, func(v string) string {
+		switch v {
+		case "SNAP":
+			// NOTE: We use dirs.CoreSnapMountDir here as the path used will be always
+			// inside the mount namespace snap-confine creates and there we will
+			// always have a /snap directory available regardless if the system
+			// we're running on supports this or not.
+			return filepath.Join(dirs.CoreSnapMountDir, snapInfo.Name(), snapInfo.Revision.String())
+		case "SNAP_DATA":
+			return snapInfo.DataDir()
+		case "SNAP_COMMON":
+			snapInfo.CommonDataDir()
+		}
+		return ""
+	})
+}
+
+func mountEntryFromLayout(layout *snap.Layout) Entry {
+	var entry Entry
+
+	mountPoint := resolveSpecialVariable(layout.Path, layout.Snap)
+	entry.Dir = mountPoint
+
+	if layout.Bind != "" {
+		mountSource := resolveSpecialVariable(layout.Bind, layout.Snap)
+		// XXX: what about ro mounts?
+		// XXX: what about file mounts, those need x-snapd.kind=file to create correctly?
```

Comment:
Can this PR land without those?

Reply:
Yes, those were not discussed during the London sprint so don't have any design yet